src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -131,4 +143,51 @@ private function newPayPalDataFromRequest( PayPalNotificationRequest $request ):
 			->setPaymentTimestamp( $request->getPaymentTimestamp() );
 	}
 
+	private function donationWasBookedWithDifferentTransactionId( Donation $donation,
+																  PayPalNotificationRequest $request ): bool {
+		/**
+		 * @var PayPalPayment $payment
+		 */
+		$payment = $donation->getPaymentMethod();
+
+		if ( !$donation->isBooked() ) {
+			return false;
+		}
+
+		if ( $request->getTransactionId() === $payment->getPayPalData()->getPaymentId() ) {
+			return false;
+		}
+
+		if ( $payment->getPayPalData()->hasChildPayment( $request->getTransactionId() ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private function createChildDonation( Donation $donation, PayPalNotificationRequest $request ) {
+		$childPaymentMethod = new PayPalPayment( $this->newPayPalDataFromRequest( $request ) );
+		$payment = $donation->getPayment();
+		$childDonation = new Donation(
+			null,
+			Donation::STATUS_EXTERNAL_BOOKED,
+			$donation->getDonor(),
+			new DonationPayment( $payment->getAmount(), $payment->getIntervalInMonths(), $childPaymentMethod ),
```

Comment:
The amount should be taken from `$request`, since it might have been changed by the donor.